pkg/systemtemplate/import.go

```diff
@@ -70,30 +72,39 @@ func toFeatureString(features map[string]bool) string {
 }
 
 func SystemTemplate(resp io.Writer, agentImage, authImage, namespace, token, url string, isWindowsCluster bool,
-	cluster *apimgmtv3.Cluster, features map[string]bool, taints []corev1.Taint, secretLister v1.SecretLister) error {
-	var tolerations, agentEnvVars string
-	d := md5.Sum([]byte(url + token + namespace))
-	tokenKey := hex.EncodeToString(d[:])[:7]
+	cluster *apimgmtv3.Cluster, features map[string]bool, tolerations []corev1.Toleration, affinity *corev1.Affinity,
```

Comment:
This function takes too many arguments as is and we're adding even more to it. 13 arguments is way too many.

Now, thinking about it further we already pass `cluster *apimgmtv3.Cluster` object here, so why can't we just extract additional tolerations / affinity (with fallback to default) / resource requirements in this func?